.github/workflows/set-date.yml

```diff
@@ -0,0 +1,85 @@
+name: Set the End Date in the project
+
+on:
+  issues:
+    types: [closed]
+jobs:
+  set_date:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Get project data
+        env:
+          GITHUB_TOKEN: ${{ secrets.GHA_TOKEN }} 
+          ORGANIZATION: keptn
+          PROJECT_NUMBER: 18
+        run: |
+          gh api graphql -f query='
+            query($org: String!, $number: Int!) {
+              organization(login: $org){
+                projectV2(number: $number) {
+                  id
+                  fields(first:20) {
+                    nodes {
+                      ... on ProjectV2Field {
+                        id
+                        name
+                      }
+                      ... on ProjectV2SingleSelectField {
+                        id
+                        name
+                        options {
+                          id
+                          name
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }' -f org=$ORGANIZATION -F number=$PROJECT_NUMBER > project_data.json
+          echo 'PROJECT_ID='$(jq '.data.organization.projectV2.id' project_data.json) >> $GITHUB_ENV
+          echo 'DATE_FIELD_ID='$(jq '.data.organization.projectV2.fields.nodes[] | select(.name== "End Date") | .id' project_data.json) >> $GITHUB_ENV
+      
+      - name: Get date
+        run: echo "DATE=$(date +"%Y-%m-%d")" >> $GITHUB_ENV
+        
+      - name: Add item id
+        env:
+          GITHUB_TOKEN: ${{ secrets.GHA_TOKEN  }}
+          ISSUE_ID: ${{ github.event.issue.node_id }}
+        run: |
+          item_id="$( gh api graphql -f query='
+            mutation($project:ID!, $issue:ID!) {
+              addProjectV2ItemById(input: {projectId: $project, contentId: $issue}) {
+                item {
+                  id
+                }
+              }
+            }' -f project=$PROJECT_ID -f issue=$ISSUE_ID --jq '.data.addProjectV2ItemById.item.id')"
+          echo 'ITEM_ID='$item_id >> $GITHUB_ENV
+                         
+      - name: Set End Date
```

Comment:
In the `Set End Date` step, the pipeline will look for an item inside our project (item refers to the issue that is triggering the workflow) whose ID matches the `ITEM_ID` and then look for a date field whose ID matches the `DATE_FIELD_ID` and set its value to `DATE`.